packages/flutter_tools/test/general.shard/build_system/targets/ios_test.dart

```diff
@@ -181,21 +181,21 @@ void main() {
     Platform: () => macPlatform,
   });
 
-  testUsingContext('AotAssemblyRelease throws exception if asked to build for x86 target', () async {
+  testUsingContext('AotAssemblyRelease throws exception if asked to build for simulator', () async {
     final FileSystem fileSystem = MemoryFileSystem.test();
     final Environment environment = Environment.test(
       fileSystem.currentDirectory,
       defines: <String, String>{
         kTargetPlatform: 'ios',
-        kSdkRoot: 'path/to/sdk',
+        kSdkRoot: 'path/to/iPhoneSimulator.sdk',
+        kBuildMode: 'release',
+        kIosArchs: 'arm64'
```

Comment:
should this still be x86_64?

Reply:
I was showing off that it didn't need to be x86_64 to throw the exception, but I changed it back to make it less confusing.